tools/raiddit/generate_claims.py

```diff
@@ -0,0 +1,112 @@
+from raiden.utils.formatting import to_checksum_address, to_hex_address
+from raiden.utils.signer import Signer
+from raiden.utils.typing import Address, Any, Dict, Optional, TokenNetworkAddress
+from raiden_contracts.utils.type_aliases import ChainID, Signature, TokenAmount
+
+DEFAULT_TOKEN_AMOUNT = TokenAmount(100)
+
+
+@dataclass
+class Claim:
+    chain_id: ChainID
+    token_network_address: TokenNetworkAddress
+    owner: Address
+    partner: Address
+    total_amount: TokenAmount
+    signature: Optional[Signature] = None
+
+    def pack(self) -> bytes:
+        return (
+            Web3.toBytes(hexstr=to_hex_address(self.token_network_address))
+            + encode_single("uint256", self.chain_id)
+            + Web3.toBytes(hexstr=to_hex_address(self.owner))
+            + Web3.toBytes(hexstr=to_hex_address(self.partner))
+            + encode_single("uint256", self.total_amount)
+        )
+
+    def sign(self, signer: Signer) -> None:
+        self.signature = signer.sign(data=self.pack())
+
+    def serialize(self) -> Dict[str, Any]:
+        assert self.signature is not None, "Claim not signed yet"
+        return dict(
+            chain_id=self.chain_id,
+            token_network_address=to_checksum_address(self.token_network_address),
+            owner=to_checksum_address(self.owner),
+            partner=to_checksum_address(self.partner),
+            signature=to_hex(self.signature),
+        )
+
+
+@click.group(name="generate")
+def generate():
+    pass
+
+
+@generate.command()
+@click.option("--address", help="Own address", required=True, type=ADDRESS_TYPE)
+@click.option("--token-network-address", required=True, type=ADDRESS_TYPE)
+@click.option("--chain-id", default=5, type=ChainID)
+@click.option("-h", "--hub-address", type=ADDRESS_TYPE)
+@click.option(
+    "-u", "--users", default=10, type=int, help="How many users claims should be generated for",
+)
+@click.option("-o", "--output-file", default=Path("./claims.json"), type=click.Path())
+def hub(
+    address: Address,
+    token_network_address: TokenNetworkAddress,
+    chain_id: ChainID,
+    hub_address: Optional[Address],
+    users: int,
+    output_file: Path,
+):
+    signer = make_signer()
+    if hub_address is None:
+        hub_address = make_address()
+
+    addresses = [address]
+    addresses.extend(make_address() for _ in range(users))
+    claims = []
+
+    for address in addresses:
+        claim = Claim(
+            chain_id=chain_id,
+            token_network_address=token_network_address,
+            owner=address,
+            partner=hub_address,
+            total_amount=DEFAULT_TOKEN_AMOUNT,
+        )
+        claim.sign(signer)
+        claims.append(claim.serialize())
+
+        reverse_claim = Claim(
+            chain_id=chain_id,
+            token_network_address=token_network_address,
+            owner=hub_address,
+            partner=address,
+            total_amount=DEFAULT_TOKEN_AMOUNT,
+        )
```

Comment:
We probably don't want to make the hub rich this way, but this is useful for testing.

Reply:
At least as an answer from reddit, they would run a hub and provide liquidity for it. Anyway, the command is specifically for creating a hub solution. We should definitely implement other commands to generate other topologies in the future.